src/hooks/v2/contractReader/V2ContractReader.ts

```diff
@@ -1,92 +1,13 @@
-import { Contract, EventFilter } from '@ethersproject/contracts'
 import { V2UserContext } from 'contexts/v2/userContext'
-import { useContractReadValue } from 'hooks/ContractReadValue'
+import { ContractReaderProps, useContractReader } from 'hooks/ContractReader'
 import { V2ContractName } from 'models/v2/contracts'
-import { useCallback, useContext } from 'react'
-import { useDeepCompareEffectNoCheck } from 'use-deep-compare-effect'
-import { getContract } from 'utils/getContract'
-
-type ContractUpdateOn = {
-  contract?: ContractConfig
-  eventName?: string
-  topics?: EventFilter['topics']
-}[]
-
-type ContractConfig = V2ContractName | Contract | undefined
+import { useContext } from 'react'
 
 export type ContractReadResult<V> = { data: V | undefined; loading: boolean }
 
-export default function useV2ContractReader<V>({
-  contract,
-  functionName,
-  args,
-  updateOn,
-  formatter,
-  callback,
-  valueDidChange,
-}: {
-  contract?: ContractConfig
-  functionName?: string
-  args?: unknown[] | null
-  updateOn?: ContractUpdateOn
-  formatter?: (val?: any) => V | undefined // eslint-disable-line @typescript-eslint/no-explicit-any
-  callback?: (val?: V) => void
-  valueDidChange?: (oldVal?: V, newVal?: V) => boolean
-}): ContractReadResult<V> {
+export default function useV2ContractReader<V>(
```

Comment:
wdyt about further abstracting the contractReader to be entirely version agnostic, and accept a ContractConfig that could be either V1 or V2?

the benefit would just be simplifying our hooks. not sure if there are any significant downsides when it comes to implementing

contracts are still read through the versioned V1/V2UserContext, so seems unlikely that a component would ever call the wrong version of a contract

Reply:
We technically can do this (it is just using the contract reader defined in this PR directly), but I don't know if the refactoring of all the code is worth it. It isn't a big job, more just tedious. 

Keeping it separate also opens our options for flexibility down the track (if we ever needed to add something V2/Vn specific) in the future

Reply:
ya makes sense 👍 if we do wanna refactor it that far we can do it in a different PR